strassengezwitscher/strassengezwitscher/models.py

```diff
@@ -20,3 +43,93 @@ def __str__(self):
+            by square_params.
+
+
+            The left bottom corner in square is called 'Min' and defined by 'min_lat' and 'min_long'.
+            The right top corner is called 'Max' and defined by 'max_lat' and 'max_long'.
+            The MapObject to check is called 'MO'.
+            'MO' is in the square iff 'MO' is right/above of 'Min' and left/below 'Max'.
+            Whenever 'Min' is left of the +180/-180 longitude and 'MO'/'Max' right of it, 'MO'/'Max' get
+            shifted by 360 degrees.
+
+             _______.< Max
+            |       |
+            |  .'MO'|
+            ._______|
+            ^
+            Min
+        """
+        square_params_decimal = cls.get_as_decimals(square_params)
+
+        if not cls.are_valid_params(square_params_decimal):
+            raise ValidationError(
+                {
+                    'message': ('Please provide latitude values in range [-90, 90] and'
+                                'longitude values in range [-180, 180].')
+                })
+
+        return [o for o in cls.objects.all() if o.is_in_square(square_params_decimal)]
+
+    @classmethod
+    def are_valid_params(cls, square_params):
+        return cls.is_valid_latitude(square_params['min_lat']) and \
+            cls.is_valid_latitude(square_params['max_lat']) and \
+            cls.is_valid_longitude(square_params['min_long']) and \
+            cls.is_valid_longitude(square_params['max_long'])
+
+    @classmethod
+    def get_as_decimals(cls, square_params):
+        try:
+            square_params_decimal = {
+                'min_lat': Decimal(square_params['min_lat']),
+                'max_lat': Decimal(square_params['max_lat']),
+                'min_long': Decimal(square_params['min_long']),
+                'max_long': Decimal(square_params['max_long'])
+            }
+
+            return square_params_decimal
+        except:
+            raise ValidationError(
+                {'message': "Please provide decimal values for 'min_lat', 'max_lat', 'min_long', 'max_long'."})
+
+    @classmethod
+    def is_valid_longitude(cls, longitude):
+        return -180 <= longitude <= 180
+
+    @classmethod
+    def is_valid_latitude(cls, latitude):
+        return -90 <= latitude <= 90
+
+    def is_in_square(self, square_params):
+        return self.is_between_latitudes(square_params['min_lat'], square_params['max_lat']) and \
+            self.is_between_longitudes(
+                square_params['min_long'], square_params['max_long'])
+
+    def is_between_longitudes(self, min_long, max_long):
+
+        location_long = self.location_long
+        if min_long > max_long:
+
+            # 'Min' [left of] +180/-180 [left of] 'Max'
+            # --> shift 'Max'
+            max_long = max_long + 360
+
+            if location_long < 0:
+                # 'Min' [left of] +180/-180 [left of] 'MO'
+                # --> shift 'MO'
+                location_long = self.location_long + 360
```

Comment:
Da du in Zeile 120 bereits `location_long` setzt, sollte
```python
location_long += 360
```
auch ok sein, oder?
In Zeile 125 könntest du auch `+=` benutzen.